crates/ruff/src/importer/mod.rs

```diff
@@ -230,12 +229,80 @@ impl<'a> Importer<'a> {
     }
 }
 
-enum Resolution {
+/// The result of an [`Importer::get_symbol`] call.
+enum GetResolution {
     /// The symbol is available for use.
-    Success(Edit, String),
+    Ok(Edit, String),
+    /// The symbol is imported, but the import came after the current location.
+    LateBinding,
+    /// The symbol is imported, but in an incompatible context (e.g., in typing-only context, while
+    /// we're in a runtime context).
+    IncompatibleContext,
+}
```

Comment:
This feels a bit off, like this should probably be called `GetResult`, but it's not an actual `Result`. I could instead make this `GetError`, and have `get_symbol` return `Option<Result<(Edit, String), GetError>> `?

Reply:
Agree. That feels odd. Why do we need it when adding a custom error type?

Reply:
Fixed.